include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp

```diff
@@ -374,17 +396,43 @@ SpatialOperator<dim, Number>::deserialize_vectors(
     }
 
     ExaDG::GridToGridProjection::GridToGridProjectionData<dim> data;
+    data.rpe_data.rtree_level            = param.restart_data.rpe_rtree_level;
     data.rpe_data.tolerance              = param.restart_data.rpe_tolerance_unit_cell;
     data.rpe_data.enforce_unique_mapping = param.restart_data.rpe_enforce_unique_mapping;
 
-    ExaDG::GridToGridProjection::do_grid_to_grid_projection<dim, Number, VectorType>(
-      checkpoint_vectors,
-      checkpoint_dof_handlers,
-      checkpoint_mapping,
-      vectors_per_dof_handler,
-      dof_handlers,
-      *matrix_free,
-      data);
+    if(param.restart_data.consider_mapping_read_target)
```

Comment:
this is the important bit:
if `consider_mapping_read_target == true`
we reuse the matrix_free object, which has the mapping setup already.
if not, we use ~do_~`(!)`grid_to_grid_projection()`, which sets up an entire matrix_free object.
I tried also with
```
matrix_free->update_mapping(*target_mapping)
[grid-to-grid interpolation]
matrix_free->update_mapping(this->get_mapping())
```
but the matrix_free object is `const` in the present context.